module/VuFind/src/VuFind/Service/MarkdownFactory.php

```diff
@@ -93,104 +111,194 @@ public function __invoke(
         $requestedName,
         array $options = null
     ) {
-        $markdownConfig = $container->get(\VuFind\Config\PluginManager::class)
-            ->get('markdown');
-
-        $environment = $this->getEnvironment($markdownConfig);
-        $this->addExtensions($environment, $markdownConfig);
+        $this->config = $container->get(\VuFind\Config\PluginManager::class)
+            ->get('markdown')->toArray();
+        $this->extensions = isset($this->config['Markdown']['extensions'])
+            ? array_map(
+                'trim',
+                explode(',', $this->config['Markdown']['extensions'])
+            )
+            : self::$defaultExtensions;
+        $this->extensions = array_filter(
+            $this->extensions,
+            function ($item) {
```

Comment:
The default behavior of array_filter is to drop empty entries. Do you need this callback? (If there's potentially a value in the array that PHP considers empty but which is not a blank string, then it is justified... but otherwise maybe the code can be simplified).

Reply:
I just wanted to be explicit here. But you are right, I can't imagine that is filtered out by default behavior could be useble as extension name. I've simplified it.